assembly/src/parsers/context.rs

```diff
@@ -251,13 +252,13 @@ impl ParserContext {
         // consume the 'end' token
         match tokens.read() {
             None => {
-                let token = tokens.read_at(proc_start).expect("no proc token");
+                let token = tokens.read_at(proc_start).expect("no end token");
```

Comment:
I think this and other messages here are actually correct. This line tries to read a token at `proc_start` which should have been a `proc` token.

Reply:
oh, thanks! I didn't notice this was reading at `proc_start`, let me check the messages again.

Reply:
fixed, it was only the `while` that was using the wrong token name, thanks for getting this!